### PR TITLE
use minimum versions for provider pinning

### PR DESCRIPTION
**Pull request by woz5999**
## what
set minimum versions for providers without pinning to a specific major version

## why
the current method makes it very difficult to maintain or upgrade consistent provider versions within a project

## references
https://www.terraform.io/docs/configuration/provider-requirements.html#best-practices-for-provider-versions

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -364,7 +364,7 @@ resource "aws_cloudfront_distribution" "default" {
 }
 
 module "dns" {
-  source           = "git::https://github.com/cloudposse/terraform-aws-route53-alias.git?ref=tags/0.8.0"
+  source           = "git::https://github.com/cloudposse/terraform-aws-route53-alias.git?ref=tags/0.9.0"
```
Comment (woz5999) [inline main.tf:367]: per https://github.com/cloudposse/terraform-aws-route53-alias/pull/25

**Comment by woz5999** [comment]
@aknysh

File: main.tf (hunk 1 above)
Comment (Nuru) [inline main.tf:367]:
```suggestion
  source           = "git::https://github.com/cloudposse/terraform-aws-route53-alias.git?ref=tags/0.8.2"
```

**Comment by Nuru** [comment]
@osterman AWS provider 3.0 has  a breaking change that breaks this module. `aws_s3_bucket` [2.x](https://registry.terraform.io/providers/hashicorp/aws/2.70.0/docs/resources/s3_bucket#region) allows setting the region on the resource, while 3.x forbids it. How do you want to handle this?

**Comment by woz5999** [comment]
@Nuru if it's helpful, i've got an aws 3.0 PR open for the s3 bucket module

**Comment by osterman** [comment]
Ya all of our bucket modules have problem like the tf state backend module. I say bite the bullet and upgrade. However the fix is just to remove the region and the region passed with the provider should work for both versions, no?

**Comment by woz5999** [comment]
yeah that's correct. and anyone with a corner case who might have been specifying a bucket region different from the provider would get an error on upgrade rather than tf accidentally trying to make some kind of destruction change

**Review by Nuru — APPROVED** [review]

**Pull request merged**